File: frontend/fix_sidebar.js
const fs = require('fs');
const glob = require('glob');

const files = glob.sync('/Users/devanshbehl/Documents/Code/codenexus2/frontend/src/pages/**/*.tsx');

files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    let changed = false;

    // Fix 1: Change "APPLICATIONS" to "PROFILE" and update path
    if (content.includes("label: 'APPLICATIONS'")) {
        content = content.replace(/label:\s*'APPLICATIONS',\s*path:\s*'\/[a-z]+\/dashboard'/g, "label: 'PROFILE', path: '/student/profile'");
        content = content.replace(/label:\s*'APPLICATIONS',\s*onClick:\s*\(\)\s*=>\s*navigate\('\/student\/dashboard'\)/g, "label: 'PROFILE', onClick: () => navigate('/student/profile')");
        changed = true;
    }
    
    // Fix 2: Check if button map exists without onClick and fix it.
    if (content.includes("sidebarItems.map((item, index) => (") && content.includes("<button\n") && !content.includes("onClick={item.onClick || (() => item.path && navigate(item.path))}")) {
        // We need to inject onClick handler into the button, and ensure 'navigate' is imported/used.
        // We'll insert onClick right after key={index}
        content = content.replace(/<button\s*key=\{index\}\s*className=/g, "<button\n                            key={index}\n                            onClick={item.onClick || (() => item.path ? navigate(item.path) : window.location.href = item.path)}\n                            className=");
        changed = true;
    }

    if (changed) {
        // Make sure navigate is available if we use it, but since they are pages they likely import useNavigate or we can just use window.location.href as fallback
        content = content.replace(/onClick=\{item\.onClick \|\| \(\(\) => item\.path \? navigate\(item\.path\) : window\.location\.href = item\.path\)\}/g, "onClick={item.onClick || (() => { if(item.path) { try { navigate(item.path) } catch(e) { window.location.href = item.path } } })}")
        fs.writeFileSync(file, content, 'utf8');
        console.log('Fixed:', file);
    }
});

File: frontend/fix_sidebar.py
import os
import glob

def fix_file(filepath):
    with open(filepath, 'r') as f:
        content = f.read()
    
    changed = False

    # Replace APPLICATIONS with PROFILE
    if "label: 'APPLICATIONS'" in content:
        content = content.replace("label: 'APPLICATIONS', path: '/student/dashboard'", "label: 'PROFILE', path: '/student/profile'")
        # Handle cases where it might use onClick
        content = content.replace("label: 'APPLICATIONS', onClick: () => navigate('/student/dashboard')", "label: 'PROFILE', onClick: () => window.location.href = '/student/profile'")
        changed = True

    # Fix the map button onClick
    if "sidebarItems.map((item, index) => (" in content and "<button" in content:
        if "onClick={item.onClick" not in content and "onClick={() => window.location.href" not in content:
            # Inject the routing logic into the button mapping
            content = content.replace("<button\n                            key={index}", "<button\n                            key={index}\n                            onClick={() => { if(item.onClick) item.onClick(); else if(item.path) window.location.href=item.path; }}\n")
            changed = True

    if changed:
        with open(filepath, 'w') as f:
            f.write(content)
        print(f"Fixed {filepath}")

for root, dirs, files in os.walk('/Users/devanshbehl/Documents/Code/codenexus2/frontend/src/pages'):
    for file in files:
        if file.endswith('.tsx'):
            fix_file(os.path.join(root, file))

